contractsV5/KyberHintHandler.sol

```diff
@@ -223,85 +184,127 @@ contract KyberHintParser is IKyberHint, Utils {
     {
         bytes memory opcode = hint.slice(indexToContinueFrom, 1);
         bytes32 opcodeHash = keccak256(opcode);
-        ReservesHint memory reserves;
+        ReservesHint memory reservesHint;
 
         if (isTokenToEth) {
-            reserves = tradeHint.tokenToEthReserves;
+            reservesHint = tradeHint.tokenToEthReserves;
         } else {
-            reserves = tradeHint.ethToTokenReserves;
+            reservesHint = tradeHint.ethToTokenReserves;
         }
 
         indexToContinueFrom += 1;
+        reservesHint.invalidHint = false;
 
-        if (opcodeHash == keccak256(END_OPCODE)) {
+        if (opcodeHash == END_KECCAK) {
             return;
-        } else if (opcodeHash == keccak256(SEPARATOR)) {
+        } else if (opcodeHash == SEPARATOR_KECCAK) {
             decodeOperation(hint, tradeHint, indexToContinueFrom, false);
-        } else if (opcodeHash == keccak256(parseOpcode(TradeType.MaskIn))) {
-            reserves.tradeType = TradeType.MaskIn;
-            (reserves.addresses, reserves.splitValuesBps, indexToContinueFrom) = decodeReservesFromHint(false, hint, indexToContinueFrom);
+        } else if (opcodeHash == MASK_IN_KECCAK) {
+            reservesHint.tradeType = TradeType.MaskIn;
+            (indexToContinueFrom) = decodeReservesFromHint(false, hint, reservesHint, indexToContinueFrom);
             decodeOperation(hint, tradeHint, indexToContinueFrom, isTokenToEth);
-        } else if (opcodeHash == keccak256(parseOpcode(TradeType.MaskOut))) {
-            reserves.tradeType = TradeType.MaskOut;
-            (reserves.addresses, reserves.splitValuesBps, indexToContinueFrom) = decodeReservesFromHint(false, hint, indexToContinueFrom);
+        } else if (opcodeHash == MASK_OUT_KECCAK) {
+            reservesHint.tradeType = TradeType.MaskOut;
+            (indexToContinueFrom) = decodeReservesFromHint(false, hint, reservesHint, indexToContinueFrom);
             decodeOperation(hint, tradeHint, indexToContinueFrom, isTokenToEth);
-        } else if (opcodeHash == keccak256(parseOpcode(TradeType.Split))) {
-            reserves.tradeType = TradeType.Split;
-            (reserves.addresses, reserves.splitValuesBps, indexToContinueFrom) = decodeReservesFromHint(true, hint, indexToContinueFrom);
+        } else if (opcodeHash == SPLIT_TRADE_KECCAK) {
+            reservesHint.tradeType = TradeType.Split;
+            (indexToContinueFrom) = decodeReservesFromHint(true, hint, reservesHint, indexToContinueFrom);
             decodeOperation(hint, tradeHint, indexToContinueFrom, isTokenToEth);
-        } else {
-            revert("Invalid hint opcode");
+        } else {            
+            reservesHint.invalidHint = true;
```

Comment:
seems better to revert from 2 reasons.
1. gives user a better idea why it reverted (if its on chain call)
2. on chain it saves user gas to revert as early as possible.
when reverting also add current index to message.

Reply:
Will think more about failingIndex. Variables cannot be added to revert reason as mentioned above, and only static string.

Reply:
at this stage can change to revert.
invalid opcode string (issued here) is better then invalid hint
later can advance with failing index.

Reply:
Changed to revert() and removed invalidHint for now.